vispy/scene/tests/test_canvas.py

```diff
@@ -60,3 +60,30 @@ def test_canvas_render(blend_func):
         else:
             # the alpha should have some transparency
             assert (rgba_result[..., 3] != 255).any()
+
+
+@requires_application()
+def test_picking_basic():
+    """Test basic picking behavior.
+
+    Based on https://github.com/vispy/vispy/issues/2107.
+
+    """
+    with TestingCanvas(size=(125, 125), show=True, title='run') as c:
+        view = c.central_widget.add_view()
+        view.margin = 5  # add empty space where there are no visuals
+        view.camera = 'panzoom'
+
+        x = np.linspace(0, 400, 100)
+        y = np.linspace(-144.1, -9.44, 100)
```

Comment:
Do these specific numbers matter? I'm not sure I can relate this directly to the picking position `(100, 25)` below.

Reply:
Oh, right. These are based on the min/max values of the data used for the first line in that picking example. Theoretically they should be arbitrary. The visuals_at position is a canvas position, not a data/array coordinate. It was just found manually as a Canvas position that has the line drawn on it.